crates/environ/src/module.rs

```diff
@@ -1144,3 +1170,19 @@ pub struct InstanceSignature {
     /// The name of what's being exported as well as its type signature.
     pub exports: IndexMap<String, EntityType>,
 }
+
+/// Type information about functions in a wasm module.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct FunctionType {
+    /// The type of this function, indexed into the global type tables for
+    /// a module compilation.
+    pub signature: SignatureIndex,
+    /// The index into the anyfunc table, if present. Note that this is
+    /// `reserved_value()` if the function does not escape from a module.
```

Comment:
What would you think about adding `is_escaping(&self) -> bool` and `non_escaping() -> Self` functions for `FunctionType`, instead of having `is_reserved_value()` and `reserved_value()` calls in lots of places? That'd make code using this more self-explanitory.